src/subdomains/supporting/payment/services/swiss-qr.service.ts

```diff
@@ -25,40 +29,95 @@ enum SupportedInvoiceLanguage {
   IT = 'IT',
 }
 
+enum SupportedSwissQRBillCurrency {
+  CHF = 'CHF',
+  EUR = 'EUR',
+}
+
+interface SwissQRBillTableData {
+  quantity: number | string;
+  description: any;
+  fiatAmount: number;
+}
+
 @Injectable()
 export class SwissQRService {
   constructor(private readonly assetService: AssetService, private readonly i18n: I18nService) {}
 
-  createQrCode(
-    amount: number,
-    currency: 'CHF',
-    reference: string,
-    bankInfo: BankInfoDto,
-    request?: TransactionRequest,
-  ): string {
-    const data = this.generateQrData(amount, currency, reference, bankInfo, request);
+  createQrCode(amount: number, currency: 'CHF', reference: string, bankInfo: BankInfoDto, userData?: UserData): string {
+    const data = this.generateQrData(amount, currency, bankInfo, reference, userData);
     return new SwissQRCode(data).toString();
   }
 
-  async createInvoice(
+  async createInvoiceFromRequest(
     amount: number,
-    currency: 'CHF',
+    currency: string,
     reference: string,
     bankInfo: BankInfoDto,
     request: TransactionRequest,
   ): Promise<string> {
-    const data = this.generateQrData(amount, currency, reference, bankInfo, request);
+    currency = Config.invoice.currencies.includes(currency) ? currency : Config.invoice.defaultCurrency;
+    if (!this.isSupportedInvoiceCurrency(currency)) {
+      throw new Error('PDF invoice is only available for CHF and EUR transactions');
+    }
+
+    const data = this.generateQrData(amount, currency, bankInfo, reference, request.userData);
+    if (!data.debtor) throw new Error('Debtor is required');
+
+    const userLanguage = request.userData.language.symbol.toUpperCase();
+    const language = this.isSupportedInvoiceLanguage(userLanguage) ? userLanguage : 'EN';
+    const asset = await this.assetService.getAssetById(request.targetId);
+    const tableData: SwissQRBillTableData = {
+      quantity: request.estimatedAmount,
+      description: {
+        assetDescription: asset.description ?? asset.name,
```

Comment:
Why do we take `asset.name` if `asset.description` is not available? 🤔 
`asset.name` is in this case duplicate with the line underneath or?